### PR TITLE
Admin Upload Component Tests

### LINKED ISSUE #487: Upload Component Test
/components/Admin/Upload

### PR DESCRIPTION
**Pull request by seanquiambao**
image

### CODE REVIEW

File: cypress/component/admin/Upload.cy.jsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
+import Upload from "@/components/admin/services/Upload";
+import { useState } from "react";
+
+describe("Upload", () => {
+  it("Upload valid mutliple files", () => {
+    const files = ["sample.pdf", "sample.jpeg", "sample.jpg", "sample.png"];
+    const Parent = () => {
+      const [email, setEmail] = useState({
+        sendto: [],
```
Comment (shahdivyank) [inline cypress/component/admin/Upload.cy.jsx]: dont need the sendto, subject or body, just focus on files

File: src/components/admin/services/Upload.jsx [hunk 2]
```diff
@@ -52,8 +52,12 @@ const Upload = ({ text, setObjects, objects, size, types }) => {
             <Col
               key={index}
               className="flex items-center bg-hackathon-gray-100 px-2 py-1 mb-1 mr-1 !max-w-fit p-0"
+              data-cy="upload-list"
             >
-              <p className="text-sm m-0 whitespace-nowrap text-gray-500 font-semibold ">
+              <p
+                className="text-sm m-0 whitespace-nowrap text-gray-500 font-semibold "
```
Comment (shahdivyank) [inline src/components/admin/services/Upload.jsx]:
```suggestion
                className="text-sm m-0 whitespace-nowrap text-gray-500 font-semibold"
```

**Review by shahdivyank — APPROVED** [review]

File: cypress/component/admin/Upload.cy.jsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
+import Upload from "@/components/admin/services/Upload";
+import { useState } from "react";
+
+describe("Upload", () => {
+  it("Upload valid mutliple files", () => {
```
Comment (shahdivyank) [inline cypress/component/admin/Upload.cy.jsx]:
```suggestion
  it("Upload Multiple Valid files", () => {
```

File: cypress/component/admin/Upload.cy.jsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        .find("input[type=file]")
+        .selectFile("cypress/fixtures/files/" + file, { force: true });
+    });
+
+    files.map((file) => {
+      cy.get(`[data-cy="${file}"]`).should("exist");
+    });
+  });
+
+  it("Upload invalid exceeding files", () => {
```
Comment (shahdivyank) [inline cypress/component/admin/Upload.cy.jsx]:
```suggestion
  it("Upload Excess Files", () => {
```

File: cypress/component/admin/Upload.cy.jsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+    cy.mount(<Parent />);
+    cy.contains("attach");
+    cy.get('[data-cy="upload-input"]')
+      .find("input[type=file]")
+      .selectFile("cypress/fixtures/files/" + file, { force: true });
+    cy.get(`[data-cy="${file}"]`).should("not.exist");
+  });
+
+  it("Upload invalid same file", () => {
```
Comment (shahdivyank) [inline cypress/component/admin/Upload.cy.jsx]:
```suggestion
  it("Upload Invalid File", () => {
```

File: cypress/component/admin/Upload.cy.jsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+      .find("input[type=file]")
+      .selectFile("cypress/fixtures/files/" + file, { force: true });
+    cy.get('[data-cy="upload-input"]')
+      .find("input[type=file]")
+      .selectFile("cypress/fixtures/files/" + file, { force: true });
+    cy.get(`[data-cy="${file}"]`).should("exist");
+    cy.get('[data-cy="upload-list"]').should("have.length", 1);
+  });
+
+  it("Remove a file", () => {
```
Comment (shahdivyank) [inline cypress/component/admin/Upload.cy.jsx]:
```suggestion
  it("Remove File", () => {
```

**Review by menthy-wu — APPROVED** [review]

**Pull request merged**